src/plugins/crypto/testmod_crypto.c

```diff
@@ -0,0 +1,222 @@
+/**
+* @file
+*
+* @brief test suite for the crypto plugin
+*
+* @copyright BSD License (see doc/COPYING or http://www.libelektra.org)
+*
+*/
+
+#ifdef HAVE_KDBCONFIG_H
+#include "kdbconfig.h"
+#endif
+#ifdef HAVE_STDLIB_H
+#include <stdlib.h>
+#endif
+#ifdef HAVE_STRING_H
+#include <string.h>
+#endif
+
+#include <stdio.h>
+#include <kdb.h>
+#include <tests_plugin.h>
+#include "crypto.h"
+
+/*
+ * The test vectors are taken from NIST SP 800-38A, section F.2.5 "CBC-AES256.Encrypt"
+ * See <http://csrc.nist.gov/publications/nistpubs/800-38a/sp800-38a.pdf> for further information.
+ */
+const unsigned char key[] =
+{
+	0x60, 0x3d, 0xeb, 0x10, 0x15, 0xca, 0x71, 0xbe,
+	0x2b, 0x73, 0xae, 0xf0, 0x85, 0x7d, 0x77, 0x81,
+	0x1f, 0x35, 0x2c, 0x07, 0x3b, 0x61, 0x08, 0xd7,
+	0x2d, 0x98, 0x10, 0xa3, 0x09, 0x14, 0xdf, 0xf4
+};
+
+const unsigned char iv[] =
+{
+	0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+	0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f
+};
+
+const unsigned char plainText[] =
+{
+	0x6b, 0xc1, 0xbe, 0xe2, 0x2e, 0x40, 0x9f, 0x96,
+	0xe9, 0x3d, 0x7e, 0x11, 0x73, 0x93, 0x17, 0x2a
+};
+
+const unsigned char expectedCipherText[] =
+{
+	0xf5, 0x8c, 0x4c, 0x04, 0xd6, 0xe5, 0xf1, 0xba,
+	0x77, 0x9e, 0xab, 0xfb, 0x5f, 0x7b, 0xfb, 0xd6
+};
+
+/**
+ * @brief checks two buffers for equality
+ *
+ * Helper function that returns zero (0) if the contents of the buffers b1 and b2 match
+ * and non-zero otherwise.
+ */
+int cmp_buffers(const unsigned char *b1, size_t len1, const unsigned char* b2, size_t len2)
```

Comment:
should be static; or even better: a macro that plays nicely with the testing framework (and prints outs where exactly the buffers differ)